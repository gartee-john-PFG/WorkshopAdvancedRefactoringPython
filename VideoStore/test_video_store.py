# from approvaltests.approvals import verify
from approvaltests.combination_approvals import verify_all_combinations
from customer import Customer, Movie, Rental

def test_what_does_customer_statement_do():
    customer_name = ["Test Customer"]
    movie_name = ["Star Wars"]
    price_code = [Movie.REGULAR, Movie.NEW_RELEASE, Movie.CHILDRENS]
    days_rented = [2, 3, 4]

    # result = make_statement(customer_name, movie_name, price_code)
    
    # verify(result)
    verify_all_combinations(make_statement,[customer_name, movie_name, price_code, days_rented])

def make_statement(customer_name, movie_name, price_code, days_rented):
    cust = Customer(customer_name)
    movie = Movie(movie_name, ((price_code + 1) % 3))
    rental = Rental(movie, days_rented)
    cust.add_rental(rental)

    movie2 = Movie(movie_name + ": The Sequal", ((price_code + 2) % 3))
    rental2 = Rental(movie2, days_rented)
    cust.add_rental(rental2)  

    movie3 = Movie(movie_name + ":Part III", ((price_code+ 3) % 3))
    rental3 = Rental(movie3, days_rented)
    cust.add_rental(rental3)

    result = cust.statement()
    return result