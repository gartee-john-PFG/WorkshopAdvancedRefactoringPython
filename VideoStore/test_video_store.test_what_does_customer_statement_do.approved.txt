args: ('Test Customer', 'Star Wars', 0, 2) => 'Rental Record for Test Customer\n\tStar Wars\t6\n\tStar Wars: The Sequal\t1.5\n\tStar Wars:Part III\t2\nAmount owed is 9.5\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 0, 3) => 'Rental Record for Test Customer\n\tStar Wars\t9\n\tStar Wars: The Sequal\t1.5\n\tStar Wars:Part III\t3.5\nAmount owed is 14.0\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 0, 4) => 'Rental Record for Test Customer\n\tStar Wars\t12\n\tStar Wars: The Sequal\t3.0\n\tStar Wars:Part III\t5.0\nAmount owed is 20.0\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 2) => 'Rental Record for Test Customer\n\tStar Wars\t1.5\n\tStar Wars: The Sequal\t2\n\tStar Wars:Part III\t6\nAmount owed is 9.5\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 3) => 'Rental Record for Test Customer\n\tStar Wars\t1.5\n\tStar Wars: The Sequal\t3.5\n\tStar Wars:Part III\t9\nAmount owed is 14.0\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 4) => 'Rental Record for Test Customer\n\tStar Wars\t3.0\n\tStar Wars: The Sequal\t5.0\n\tStar Wars:Part III\t12\nAmount owed is 20.0\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 2) => 'Rental Record for Test Customer\n\tStar Wars\t2\n\tStar Wars: The Sequal\t6\n\tStar Wars:Part III\t1.5\nAmount owed is 9.5\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 3) => 'Rental Record for Test Customer\n\tStar Wars\t3.5\n\tStar Wars: The Sequal\t9\n\tStar Wars:Part III\t1.5\nAmount owed is 14.0\nYou earned 4 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 4) => 'Rental Record for Test Customer\n\tStar Wars\t5.0\n\tStar Wars: The Sequal\t12\n\tStar Wars:Part III\t3.0\nAmount owed is 20.0\nYou earned 4 frequent renter points'
